rust/src/connection/network/transmitter/transaction.rs

```diff
@@ -244,22 +286,20 @@ impl TransactionTransmitter {
         };
         tokio::task::spawn_blocking({
             let collector = collector.clone();
-            move || {
-                Self::dispatch_loop(queue_source, request_sink, collector, on_close_callback_source, shutdown_signal)
-            }
+            move || Self::sync_dispatch_loop(queue_source, request_sink, collector, shutdown_signal)
         });
-        tokio::spawn(Self::listen_loop(response_source, collector, shutdown_sink));
+        tokio::spawn(Self::async_listen_loop(response_source, collector, on_close_callback_source, shutdown_sink));
```

Comment:
Since we just have to get the on_close_callback into the Collector in some way, we can choose whether we lsiten to the channel in the async listen loop or the sync dispatch loop. To minimize waits, we're going to let Tokio manage it in the Async loop.